lib/StateMachineSkill.js

```diff
@@ -34,6 +34,15 @@ class StateMachineSkill extends AlexaSkill {
       return this.requestHandlers.IntentRequest(alexaEvent, reply);
     }, true);
 
+    // we treat onDisplay.ElementSelected as an intentRequest
+    this['onDisplay.ElementSelected']((alexaEvent, reply) => {
+      const intent = 'DisplayElementSelected';
+      _.set(alexaEvent, 'intent.name', intent);
+      _.set(alexaEvent, 'intent.slots', {});
+
+      return this.requestHandlers.IntentRequest(alexaEvent, reply);
```

Comment:
Again, I'm worried about making this an intent. If it stays it's own type, we can help with giving a better interface to things like what was clicked, kinda like we provide a better interface to the slots for intents.

Reply:
I provide a new way to access this token -> AlexaEvent.token